bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/internal/i18n/I18nConfigOptionsProvider.java

```diff
@@ -23,29 +28,65 @@
  *
  * @author Simon Kaufmann - initial contribution and API
+ * @author Erdoan Hadzhiyusein - Added time zone
  *
  */
 @Component(immediate = true)
 public class I18nConfigOptionsProvider implements ConfigOptionProvider {
 
+    private static final String NO_OFFSET_FORMAT = "(GMT) %s";
+    private static final String NEGATIVE_OFFSET_FORMAT = "%s (GMT%d:%02d)";
+    private static final String POSITIVE_OFFSET_FORMAT = "%s (GMT+%d:%02d)";
+
     @Override
     public Collection<ParameterOption> getParameterOptions(URI uri, String param, Locale locale) {
         if (uri.toString().equals("system:i18n")) {
             Locale translation = locale != null ? locale : Locale.getDefault();
-            if (param.equals("language")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
-            } else if (param.equals("region")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
-            } else if (param.equals("variant")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+            switch (param) {
+                case "language":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
+                case "region":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
+                case "variant":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+                case "timezone":
+                    Collection<ParameterOption> collection = new LinkedList<ParameterOption>();
+                    String[] ids = TimeZone.getAvailableIDs();
+                    List<TimeZone> timeZones = Arrays.asList(ids).stream().map(TimeZone::getTimeZone)
+                            .collect(Collectors.toList());
+                    Collections.sort(timeZones, (s1, s2) -> s2.getRawOffset() - s1.getRawOffset());
+                    // sorting the time zones by offset
+                    for (TimeZone instance : timeZones) {
+                        collection.add(new ParameterOption(instance.getID(), getTimeZoneRepresentation(instance)));
+                    }
+                    return collection;
+                default:
+                    return null;
             }
         }
         return null;
     }
 
+    private static String getTimeZoneRepresentation(TimeZone tz) {
+        long hours = TimeUnit.MILLISECONDS.toHours(tz.getRawOffset());
+        long minutes = TimeUnit.MILLISECONDS.toMinutes(tz.getRawOffset()) - TimeUnit.HOURS.toMinutes(hours);
+        minutes = Math.abs(minutes);
+        String result = "";
```

Comment:
Use `final String result;` so you ensure that result needs to be assigned in one of the branches and you don't need to assign it twice.